examples/pa_engine_single_unit_example.py

```diff
@@ -0,0 +1,125 @@
+from fds.analyticsapi.engines.api.components_api import ComponentsApi
+from fds.analyticsapi.engines.api_client import ApiClient
+from fds.analyticsapi.engines.configuration import Configuration
+from fds.analyticsapi.engines.model.component_summary import ComponentSummary
+from fds.analyticsapi.engines.model.pa_calculation_parameters_root import PACalculationParametersRoot
+from fds.analyticsapi.engines.model.pa_calculation_parameters import PACalculationParameters
+from fds.analyticsapi.engines.model.pa_date_parameters import PADateParameters
+from fds.analyticsapi.engines.model.pa_identifier import PAIdentifier
+from fds.protobuf.stach.extensions.StachVersion import StachVersion
+from fds.protobuf.stach.extensions.StachExtensionFactory import StachExtensionFactory
+
+from urllib3 import Retry
+
+host = "https://api.factset.com"
+username = os.environ["ANALYTICS_API_USERNAME_SERIAL"]
+password = os.environ["ANALYTICS_API_PASSWORD"]
+
+
+def main():
+    config = Configuration()
+    config.host = host
+    config.username = username
+    config.password = password
+    # add proxy and/or disable ssl verification according to your development environment
+    # config.proxy = "<proxyUrl>"
+    config.verify_ssl = False
+
+    # Setting configuration to retry api calls on http status codes of 429 and 503.
+    config.retries = Retry(total=3, status=3, status_forcelist=frozenset([429, 503]), backoff_factor=2,
+                           raise_on_status=False)
+
+    api_client = ApiClient(config)
+
+    components_api = ComponentsApi(api_client)
+
+    try:
+        pa_document_name = "PA_DOCUMENTS:DEFAULT"
+        pa_component_name = "Weights"
+        pa_component_category = "Weights / Exposures"
+        pa_benchmark_sp_50 = "BENCH:SP50"
+        pa_benchmark_r_1000 = "BENCH:R.1000"
+        startdate = "20180101"
+        enddate = "20181231"
+        frequency = "Monthly"
+
+        components = components_api.get_pa_components(pa_document_name)
+        component_summary = ComponentSummary(name=pa_component_name, category=pa_component_category)
+        component_id = [id for id in list(components.data.keys()) if components.data[id] == component_summary][0]
+        print("PA Component Id: " + component_id)
+        pa_accounts = [PAIdentifier(id=pa_benchmark_sp_50)]
+        pa_benchmarks = [PAIdentifier(id=pa_benchmark_r_1000)]
+        pa_dates = PADateParameters(startdate=startdate, enddate=enddate, frequency=frequency)
+
+        pa_calculation_parameters = {"1": PACalculationParameters(componentid=component_id, accounts=pa_accounts,
+                                                                  benchmarks=pa_benchmarks, dates=pa_dates)}
+
+        pa_calculation_parameter_root = PACalculationParametersRoot(data=pa_calculation_parameters)
+
+        pa_calculations_api = PACalculationsApi(api_client)
+
+        post_and_calculate_response = pa_calculations_api.post_and_calculate(
```

Comment:
Can you please pass the deadline header value here?

Reply:
What value would you want it to be set as in this example?

Reply:
I'll just do 30 seconds

Reply:
@saigiridhar21 since client usually copy paste these examples in their code should we add the headers to examples? I am afraid they will permanently add it even if they don't need it. Would be better if clients use the default value instead of they hard coding it here?

Reply:
@a-ghag we could just assign the default values to the header in the code and mention the same in a comment for the clients' information. Would there be an issue this way?

Reply:
So if we assign the the same default value as in our code here and later we change the default value in our code clients would still be using the old default as they would be using the hard coded value from these examples. Is it good to assume that we did this in Java? If yes, can we remove those?

Reply:
@a-ghag I am fine removing `cache-control` from the examples. Shall we document it in PDF documentation? That way clients can refer to it there.
@AkankshaAcharya Please remove the cache-control option when you work on Java SDK next time.

Reply:
Rose is working on a document which goes in detail about about cache-control directive